### PR TITLE
Type conversion fixes

### PR DESCRIPTION
**Pull request by SparkiDev**
# Description

Changes to get compilation with -Wconversion passing on the files.

# Testing

./configure --disable-shared --enable-all CFLAGS=-Wconversion
Compile the files changed only.

# Checklist

 - [ ] added tests
 - [ ] updated/added doxygen
 - [ ] updated appropriate READMEs
 - [ ] Updated manual and documentation

### CODE REVIEW

**Comment by SparkiDev** [comment]
retest this pleae

**Comment by SparkiDev** [comment]
retest this please

**Comment by SparkiDev** [comment]
retest this please

File: wolfcrypt/src/pkcs7.c [hunk 1]
```diff
@@ -14351,7 +14393,7 @@ int wc_PKCS7_SetStreamMode(PKCS7* pkcs7, byte flag,
         return BAD_FUNC_ARG;
     }
 #ifdef ASN_BER_TO_DER
-    pkcs7->encodeStream = flag;
+    pkcs7->encodeStream = (flag == 1);
```
Comment (douzzer) [inline wolfcrypt/src/pkcs7.c:14396]: `!= 0`

File: wolfcrypt/src/pkcs7.c [hunk 2]
```diff
@@ -14387,7 +14429,7 @@ int wc_PKCS7_SetNoCerts(PKCS7* pkcs7, byte flag)
     if (pkcs7 == NULL) {
         return BAD_FUNC_ARG;
     }
-    pkcs7->noCerts = flag;
+    pkcs7->noCerts = (flag == 1);
```
Comment (douzzer) [inline wolfcrypt/src/pkcs7.c:14432]: `!= 0`

File: wolfcrypt/src/pkcs7.c [hunk 3]
```diff
@@ -8194,13 +8207,15 @@ static int wc_PKCS7_EncryptContent(PKCS7* pkcs7, int encryptOID, byte* key,
                         ret = wc_AesGcmEncrypt(aes, out, in, inSz, iv, ivSz,
```
Comment (douzzer) [inline wolfcrypt/src/pkcs7.c:8207]: I'm seeing
```
wolfcrypt/src/pkcs7.c:8207:62: error: conversion to 'word32' {aka 'unsigned int'} from 'int' may change the sign of the result [-Werror=sign-conversion]
 8207 |                         ret = wc_AesGcmEncrypt(aes, out, in, inSz, iv, ivSz,
      |                                                              ^~~~
wolfcrypt/src/pkcs7.c:8207:72: error: conversion to 'word32' {aka 'unsigned int'} from 'int' may change the sign of the result [-Werror=sign-conversion]
 8207 |                         ret = wc_AesGcmEncrypt(aes, out, in, inSz, iv, ivSz,
      |                                                                        ^~~~
```

**Review by douzzer — CHANGES_REQUESTED** [review]
Good amount of progress here, though of course it would be nicer if we just had self-consistent types in the first place, rather than an armada of casts to support legacy code and API signatures.

checkpoint for the record: on a default build of this PR, I count 150 remaining `-Wconversions` in the TLS layer, 39 in `api.c`, and 28 in the crypto layer, all but 2 in `test.c`.  (and I've noted those 2 in my review.)

File: wolfssl/wolfcrypt/asn.h [hunk 4]
```diff
@@ -670,7 +670,7 @@ WOLFSSL_LOCAL void SetASN_OID(ASNSetData *dataASN, int oid, int oidType);
     do {                                                               \
         int ii;                                                        \
         for (ii = (node) + 1; ii < (int)(dataASNLen); ii++) {          \
-            if ((asn)[ii].depth <= (asn)[node].depth)                  \
+            if ((asn)[ii].depth <= (asn)[(node)].depth)                \
```
[the remark this answers is not in the document]
Comment (SparkiDev) [reply wolfssl/wolfcrypt/asn.h]: fixed

File: wolfssl/wolfcrypt/asn.h [hunk 5]
```diff
@@ -689,7 +689,7 @@ WOLFSSL_LOCAL void SetASN_OID(ASNSetData *dataASN, int oid, int oidType);
         int ii;                                                        \
         (dataASN)[node].noOut = 1;                                     \
         for (ii = (node) + 1; ii < (int)(dataASNLen); ii++) {          \
-            if ((asn)[ii].depth <= (asn)[node].depth)                  \
+            if ((asn)[ii].depth <= (asn)[(node)].depth)                \
```
[the remark this answers is not in the document]
Comment (SparkiDev) [reply wolfssl/wolfcrypt/asn.h]: fixed

File: wolfssl/wolfcrypt/asn.h [hunk 6]
```diff
@@ -594,7 +594,7 @@ WOLFSSL_LOCAL void SetASN_OID(ASNSetData *dataASN, int oid, int oidType);
  * @param [in] oidType  Type of OID.
  */
 #define SetASN_OID(dataASN, oid, oidType)                              \
-    (dataASN)->data.buffer.data = OidFromId(oid, oidType,              \
+    (dataASN)->data.buffer.data = OidFromId((oid), (oidType),          \
```
[the remark this answers is not in the document]
Comment (SparkiDev) [reply wolfssl/wolfcrypt/asn.h]: fixed

File: wolfcrypt/src/pkcs7.c [hunk 7]
```diff
@@ -3428,7 +3436,7 @@ int wc_PKCS7_SetDetached(PKCS7* pkcs7, word16 flag)
     if (pkcs7 == NULL || (flag != 0 && flag != 1))
         return BAD_FUNC_ARG;
 
-    pkcs7->detached = flag;
+    pkcs7->detached = (flag == 1);
```
[the remark this answers is not in the document]
Comment (SparkiDev) [reply wolfcrypt/src/pkcs7.c]: fixed

File: wolfcrypt/src/pkcs7.c [hunk 8]
```diff
@@ -5513,10 +5522,10 @@ static int PKCS7_VerifySignedData(PKCS7* pkcs7, const byte* hashBuf,
         #endif /* !NO_PKCS7_STREAM */
 
             /* Skip the set. */
-            idx += length;
+            idx += (word32)length;
             degenerate = (length == 0) ? 1 : 0;
         #ifndef NO_PKCS7_STREAM
-            pkcs7->stream->degenerate = degenerate;
+            pkcs7->stream->degenerate = (degenerate == 1);
```
[the remark this answers is not in the document]
Comment (SparkiDev) [reply wolfcrypt/src/pkcs7.c]: fixed

File: wolfcrypt/src/pkcs7.c [hunk 9]
```diff
@@ -5716,11 +5725,12 @@ static int PKCS7_VerifySignedData(PKCS7* pkcs7, const byte* hashBuf,
...
                     if (ret == 0) {
                         /* Use single OCTET_STRING directly, or reset length. */
-                        if (localIdx - start + length == (word32)contentLen) {
+                        if (localIdx - start + (word32)length ==
+                                                           (word32)contentLen) {
                             multiPart = 0;
                         } else {
                         #ifndef NO_PKCS7_STREAM
-                            pkcs7->stream->multi         = multiPart;
+                            pkcs7->stream->multi         = (multiPart == 1);
```
[the remark this answers is not in the document]
Comment (SparkiDev) [reply wolfcrypt/src/pkcs7.c]: fixed

File: wolfcrypt/src/pkcs7.c [hunk 10]
```diff
@@ -5748,7 +5758,7 @@ static int PKCS7_VerifySignedData(PKCS7* pkcs7, const byte* hashBuf,
                         ret = ASN_PARSE_E;
                 #ifndef NO_PKCS7_STREAM
                     if (ret == 0) {
-                        pkcs7->stream->multi         = multiPart;
+                        pkcs7->stream->multi         = (multiPart == 1);
```
[the remark this answers is not in the document]
Comment (SparkiDev) [reply wolfcrypt/src/pkcs7.c]: fixed

File: wolfcrypt/src/pkcs7.c [hunk 11]
```diff
@@ -5800,7 +5810,7 @@ static int PKCS7_VerifySignedData(PKCS7* pkcs7, const byte* hashBuf,
 
         #ifndef NO_PKCS7_STREAM
             /* save detached flag value */
-            pkcs7->stream->detached = detached;
+            pkcs7->stream->detached = (detached == 1);
```
[the remark this answers is not in the document]
Comment (SparkiDev) [reply wolfcrypt/src/pkcs7.c]: fixed

File: wolfcrypt/src/pkcs7.c [hunk 12]
```diff
@@ -6231,7 +6241,7 @@ static int PKCS7_VerifySignedData(PKCS7* pkcs7, const byte* hashBuf,
                         }
 
                         /* Restore content is PKCS#7 flag */
-                        pkcs7->contentIsPkcs7Type = contentIsPkcs7Type;
+                        pkcs7->contentIsPkcs7Type = (contentIsPkcs7Type == 1);
```
[the remark this answers is not in the document]
Comment (SparkiDev) [reply wolfcrypt/src/pkcs7.c]: fixed

File: wolfcrypt/src/pkcs7.c (hunk 1 above)
Comment (SparkiDev) [reply wolfcrypt/src/pkcs7.c]: fixed

File: wolfcrypt/src/pkcs7.c (hunk 2 above)
Comment (SparkiDev) [reply wolfcrypt/src/pkcs7.c]: fixed

**Comment by SparkiDev** [comment]
retest this please

**Review by douzzer — APPROVED** [review]

**Pull request merged**